client/landing/stepper/declarative-flow/new-hosted-site-flow.ts

```diff
@@ -55,11 +70,16 @@ const hosting: Flow = {
 
 		const query = useQuery();
 		const queryParams = Object.fromEntries( query );
+		const plan = queryParams.plan;
 		const flowName = this.name;
+		const showDomainStep = useShowDomainStep();
 
 		const goBack = () => {
 			if ( _currentStepSlug === 'plans' ) {
-				return window.location.assign( '/sites?hosting-flow=true' );
+				if ( ! showDomainStep ) {
+					return window.location.assign( '/sites?hosting-flow=true' );
+				}
+				return navigate( 'domains' );
```

Comment:
```suggestion
				if ( showDomainStep ) {
					return navigate( 'domains' );flow=true' );
				}
				return window.location.assign( '/sites?hosting-
```

Reply:
Agreed, changing it 👍

Reply:
Changed here [6f4baca](https://github.com/Automattic/wp-calypso/pull/95721/commits/6f4bacaf7fda152d23165b0071607aae1579bb1e)